covquestions-js/src/questionnaireEngine.test.ts

```diff
@@ -118,26 +119,103 @@ describe("questionnaireEngine", () => {
+          },
         },
       ],
     };
 
-    const engine = new QuestionnaireEngine(testQuestionnaire);
-    expect(engine.getProgress()).toEqual(0);
+    it("normal progress", () => {
+      const engine = new QuestionnaireEngine(testQuestionnaire);
+      expect(engine.getProgress()).toEqual(0);
+
+      const question1 = engine.nextQuestion();
+      engine.setAnswer("question1id", true);
+      expect(engine.getProgress()).toEqual(1 / 3);
+      expect(question1?.id).toEqual(question1id);
+
+      const question2 = engine.nextQuestion();
+      engine.setAnswer("question2id", true);
+      expect(engine.getProgress()).toEqual(2 / 3);
+      expect(question2?.id).toEqual(question2id);
+
+      const question3 = engine.nextQuestion();
+      engine.setAnswer("question3id", true);
+      expect(engine.getProgress()).toEqual(1);
+      expect(question3?.id).toEqual(question3id);
+
+      const afterLastQuestion = engine.nextQuestion();
+      expect(afterLastQuestion).toEqual(undefined);
+    });
+
+    it("progress - if last question disabled", () => {
+      const engine = new QuestionnaireEngine(testQuestionnaire);
+      expect(engine.getProgress()).toEqual(0);
+
+      const question1 = engine.nextQuestion();
+      engine.setAnswer("question1id", true);
+      expect(engine.getProgress()).toEqual(1 / 3);
+      expect(question1?.id).toEqual(question1id);
+
+      const question2 = engine.nextQuestion();
+      engine.setAnswer("question2id", false);
+      expect(engine.getProgress()).toEqual(1);
+      expect(question2?.id).toEqual(question2id);
+
+      const afterLastQuestion = engine.nextQuestion();
+      expect(afterLastQuestion).toEqual(undefined);
+    });
 
-    const question1 = engine.nextQuestion();
-    engine.setAnswer("question1id", true);
-    expect(engine.getProgress()).toEqual(0.5);
-    expect(question1?.id).toEqual(question1id);
+    it("progress - if almost last question disabled", () => {
+      const testQuestionnaire: Questionnaire = {
+        ...emptyTestQuestionnaire,
+        questions: [
+          {
+            id: question1id,
+            text: "Frage 1?",
+            type: "boolean",
+          },
+          {
+            id: question2id,
+            text: "Frage 2?",
+            type: "boolean",
+            enableWhenExpression: {
+              var: question1id,
+            },
+          },
+          {
+            id: question3id,
+            text: "Frage 3?",
+            type: "boolean",
+          },
+        ],
+      };
+      const engine = new QuestionnaireEngine(testQuestionnaire);
+      expect(engine.getProgress()).toEqual(0);
 
-    const question2 = engine.nextQuestion();
-    engine.setAnswer("question2id", 123);
-    expect(engine.getProgress()).toEqual(1);
-    expect(question2?.id).toEqual(question2id);
+      const question1 = engine.nextQuestion();
+      engine.setAnswer("question1id", false);
+      expect(engine.getProgress()).toEqual(1 / 3);
```

Comment:
Would we expect 1/3, 1/2 or 2/3 here?
- we gave 1 answer out of 3 total questions
- we gave 1 answer out of 2 questions that have been shown and might be shown later
- we are currently at question 3 => 2 out of 3 questions are done

I would tend to the third option: progress = (nextQuestionIndex-1)/numberOfQuestions   (questions indexed from 1 to 3)

Reply:
I think the first one is the easiest for now. Also, it better describes that only one question is answered yet.

Reply:
Ok for me